### PR TITLE
Dependency Injection

### PR DESCRIPTION
**Pull request by DavidBadura**
Before this pull request we need to register gadget, listener, subscriber, formatter, importer etc. several times (simpspector & here). with this change we dont need this anymore :wink:

### CODE REVIEW

**Comment by DavidBadura** [comment]
@simpspector/owners ready for review :wink:

File: src/Console/Application.php [hunk 1]
```diff
@@ -41,14 +44,19 @@ public function __construct(Analyser $analyser, RepositoryInterface $repository,
      */
     public static function create($bin = null)
     {
-        $buider = (new AnalyserBuilder())
-            ->setBinaryDir($bin);
+        $bin = $bin ? rtrim($bin, '/') . '/' : '';
```
Comment (larsborn) [inline src/Console/Application.php]: extract this to a StringUtil or PathHelper?

**Comment by larsborn** [comment]
I didn't even notice, that we were missing DI.

+1 for merge

**Comment by tolry** [comment]
:+1: 

Is there a reason for using yml as DI config format? Since I like this better, I am not complaining. But this seems at least not to be "best practice", if symfony standard edition is any indication.

**Comment by DavidBadura** [comment]
Yaml is the new best practice in symfony for di configuration ;)

**Comment by DavidBadura** [comment]
so, it can be merged?

File: src/Console/Application.php (hunk 1 above)
Comment (DavidBadura) [inline src/Console/Application.php]: we need this only on one place, so i think we don't need to extract this.

Comment (DavidBadura) [inline src/Console/Application.php]: btw. we should replace the PathHelper with https://github.com/webmozart/path-util & https://github.com/webmozart/glob :wink:

**Comment by DavidBadura** [comment]
@simpspector/owners ping

**Comment by tolry** [comment]
justification accepted :smile:

**Pull request merged**